src/v/archival/ntp_archiver_service.h

```diff
@@ -204,7 +204,7 @@ class ntp_archiver {
     ss::lowres_clock::duration _cloud_storage_initial_backoff;
     ss::lowres_clock::duration _segment_upload_timeout;
     ss::lowres_clock::duration _manifest_upload_timeout;
-    ss::semaphore _mutex{1};
+    ssx::semaphore _mutex{1, "archive/ntp"};
```

Comment:
Please switch the include

Reply:
Is it possible to use the actual ntp name here? Because these objects are created per ntp.

Reply:
I agree this would be nice. I've added this to the list of things to look at for followup pr #5562